specification/keyvault/Security.KeyVault.BackupRestore/models.tsp

```diff
@@ -0,0 +1,244 @@
+import "@typespec/http";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "../Security.KeyVault.Common/common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace KeyVault;
+
+/**
+ * An authentication method and location for the operation.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Model name has already shipped"
+model SASTokenParameter {
+  /**
+   * Azure Blob storage container Uri
+   */
+  storageResourceUri: string;
+
+  /**
+   * The SAS token pointing to an Azure Blob storage container
+   */
+  token?: string;
+
+  /**
+   * Indicates which authentication method should be used. If set to true, Managed HSM will use the configured user-assigned managed identity to authenticate with Azure Storage. Otherwise, a SAS token has to be specified.
+   */
+  useManagedIdentity?: boolean = false;
+}
+
+/**
+ * The status of a long-running operation.
+ */
+@lroStatus
+union OperationStatus {
+  string,
+
+  /** The operation is in progress. */
+  InProgress: "InProgress",
+
+  /** The operation successfully completed. */
+  Succeeded: "Succeeded",
+
+  /** The operation was canceled. */
+  Canceled: "Canceled",
+
+  /** The operation failed. */
+  Failed: "Failed",
+}
+
+/**
+ * Full backup operation
+ */
+model FullBackupOperation {
+  /**
+   * Status of the backup operation.
+   */
+  status?: OperationStatus;
+
+  /**
+   * The status details of backup operation.
+   */
+  statusDetails?: string;
+
+  /**
+   * Error encountered, if any, during the full backup operation.
+   */
+  error?: Error;
+
+  /**
+   * The start time of the backup operation in UTC
+   */
+  @encode("unixTimestamp", int32)
+  startTime?: utcDateTime;
+
+  /**
+   * The end time of the backup operation in UTC
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "Need to mark this as nullable"
+  @OpenAPI.extension("x-nullable", true)
```

Comment:
This makes me concerned since the swagger/openapi and the typespec now have different semantic meanings. In the TypeSpec, we are explicitly saying that a `null` value is not allowed whereas in the swagger, we explicitly say that it is.

Reply:
Is there a way to make values nullable in TSP that we're missing here? I've tried searching through the documentation and couldn't find anything